src/main/java/com/google/ehub/utility/NextIncreasingLexicographicStringUtility.java

```diff
@@ -0,0 +1,37 @@
+package com.google.ehub.utility;
+
+/**
+ * Utility class that holds method to find the next lexicographic string.
+ */
+public final class NextIncreasingLexicographicStringUtility {
+  /**
+   * Finds the next increasing lexicographic string.
+   *
+   * @param str the string used to find the next increasing lexicographic string
+   * @return the next lexicographic string in increasing order
+   */
+  public static String getNextIncreasingLexicographicString(String str) {
+    if (str == null) {
+      return "";
+    }
+    if (str.isEmpty()) {
+      return "a";
+    }
+
+    int i = str.length() - 1;
+
+    while (i >= 0 && str.charAt(i) == 'z') {
+      --i;
+    }
+
+    if (i == -1) {
+      str += 'a';
```

Comment:
behavior is a little inconsistent here because in some cases you are returning a brand new String and in some cases you're modifying the input. Usually the former is better

Reply:
I changed the method so it uses the same consistent behavior of returning a new string.